### PR TITLE
Fixing permissions in DM for bot owners

### PR DESCRIPTION
**Pull request by taOsemc**
# PULL REQUEST

## **Overview**

Fixing the permission issue in DM. 

Checking if staff is needed and if user if included in.

## **Status**

- [  ] This PR have been tested and is ready to be merged.

## **Semantic versioning classification**

- [X] PATCH: This PR fixes a bug, if needed it also references the relevant issue or documentation.

### CODE REVIEW

File: src/Utility/Discord/Prompt.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ class Prompt {
             this._emitter.once('invalidEnd', () => reject(this._onInvalidEnd() ) ); // When the prompt ends as "invalid"
 
             // listen to the even, start the prompt
-            this.client.on('messageCreate', this._boundEvent); // Bind the event
+            this.client.on(this.axon.library.enums.EVENTS.MESSAGE_CREATE, this._boundEvent); // Bind the event
```
Comment (Khaaz) [inline src/Utility/Discord/Prompt.js:177]: I'd rather have this in a separate PR. For the record I'm planning to change entirely the Prompt implementation to use Collectors but I can backport this in the meantime.
Out of the scope of this particular PR though.

File: src/Core/Command/Command.js [hunk 2]
```diff
@@ -202,14 +202,22 @@ class Command extends Base {
 
         const userID = this.library.message.getAuthorID(msg);
         const channel = this.library.message.getChannel(msg);
-
+        let canExecute = '';
```
Comment (Khaaz) [inline src/Core/Command/Command.js:205]: you don't have to hoist `canExecute`. You can just declare it both time where you need it since it's in 2 different scope.

File: src/Core/Command/Command.js [hunk 3]
```diff
@@ -223,7 +231,7 @@ class Command extends Base {
 
             /* Permissions checkers */
             if (!env.isAdmin) {
```
Comment (Khaaz) [inline src/Core/Command/Command.js:233]: if I'm not mistaken, `isAdmin` is solely based on the `id` of the caller against `staff.admins` Running the execution as admin or owner has more advantages than just bypassing permissions (typically: no cooldown) and you definitely can run as admin in DMs.

That said, I'd say that this whole block can be moved outside of the `REGULAR EXECUTION` block statement.
That means that we don't need the added lines 213-220 since we can check for canExecute only once.
In fact, that's why we add the override in side the `canExecute` function.
I'd be ok to do as you did if we were calling a different method to specifically check for bot staff but prefer the solution above.

TLDR: this whole block can be moved outside of the `regular execution` else statement.
And the `this.permissions.canExecute` above can be removed entirely

**Review by Khaaz — CHANGES_REQUESTED** [review]

**Review by Khaaz — APPROVED** [review]
